### PR TITLE
cmd/web: add markdown processing

### PR DESCRIPTION
**Pull request by Xe**
This uses the library blackfriday[1] to convert markdown to HTML. This is a naive implementation, but it will do for now.

[1]: https://pkg.go.dev/github.com/russross/blackfriday

### CODE REVIEW

File: cmd/web/main.go [hunk 1]
```diff
@@ -529,13 +575,32 @@ WHERE p.id = ?1`
...
+		return "", false
+	}
+
+	sp := strings.SplitN(input, "\n", 2)
+
+	if len(sp) != 2 {
+		return "", false
+	}
+
+	titleLine, _ := strings.CutPrefix(sp[0], "#")
```
[the remark this answers is not in the document]
Comment (shayne) [reply cmd/web/main.go:597]: This whole thing could be:
```go
title, ok := strings.CutPrefix(strings.Split(strings.TrimSpace(input), "\n")[0], "#")
```

File: cmd/web/tmpl/fancypost.tmpl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        <link rel="stylesheet" href="/static/css/font-files/inter.css" />
+        <link rel="stylesheet" href="/static/css/normalize.css" />
+        <link rel="stylesheet" href="/static/css/prism.css" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <style>
+            html {
+                background: #fff;
+                color: #333;
+                font: 16px/1.4 OpenDyslexic, Inter, Helvetica, Arial, sans-serif;
+
```
Comment (shayne) [inline cmd/web/tmpl/fancypost.tmpl:14]: empty line

File: flake.nix [hunk 3]
```diff
@@ -22,7 +22,9 @@
       let
         pkgs = import nixpkgs {
           inherit system;
-          overlays = [ gomod2nix.overlays.default ];
+          overlays = [ gomod2nix.overlays.default (final: prev: {
+            go = prev.go_1_20;
```
Comment (shayne) [inline flake.nix:26]: nice!

File: cmd/web/main.go [hunk 4]
```diff
@@ -480,6 +483,27 @@ WHERE p.id = ?1`
...
+	}
+
+	var rawHTML *template.HTML
+
+	var cssClass string
+	if lang != "" {
+		cssClass = fmt.Sprintf("lang-%s", strings.ToLower(lang))
+	}
+
+	if lang == "Markdown" || filepath.Ext(fname) == ".markdown" {
```
[the remark this answers is not in the document]
Comment (Xe) [reply cmd/web/main.go:498]: that is caught by the initial step, `.markdown` was an overly defensive addition, but I will check if it's still needed

[the remark this answers is not in the document]
Comment (Xe) [reply cmd/web/main.go:498]: Turns out I found a bug in enry! https://github.com/go-enry/go-enry/pull/154

**Review by shayne — APPROVED** [review]
🚢

**Pull request merged**